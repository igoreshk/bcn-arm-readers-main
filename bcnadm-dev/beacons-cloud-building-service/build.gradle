plugins {
    id 'beacons.common-service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-starter-oauth2')
    implementation('org.springframework.amqp:spring-rabbit')
    implementation('org.apache.commons:commons-math3:3.6.1')
    implementation('org.geotools:gt-main:26.1')
}

jacocoTestCoverageVerification {
    violationRules {
        rule {
            limit {
                minimum = 0.80
            }
        }
    }
}
