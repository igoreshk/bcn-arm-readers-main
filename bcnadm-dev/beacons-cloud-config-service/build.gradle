plugins {
    id 'beacons.service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-config-server')
    implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-netflix-eureka-client'
}
