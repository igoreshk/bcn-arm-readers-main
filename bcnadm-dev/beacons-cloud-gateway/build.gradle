plugins {
    id 'beacons.service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-starter-config')
    implementation('org.springframework.cloud:spring-cloud-starter-netflix-eureka-client')
    implementation('org.springframework.cloud:spring-cloud-starter-openfeign')
    implementation('org.springframework.boot:spring-boot-starter-web') {
        exclude group: 'org.springframework.boot', module: 'spring-boot-starter-tomcat'
    }

    implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-oauth2'
    implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-netflix-zuul'
    implementation('org.springframework.cloud:spring-cloud-starter-security')

    compile "javax.servlet:javax.servlet-api:4.0.0"

    testImplementation('org.springframework.boot:spring-boot-starter-test')
    testImplementation('org.springframework.security:spring-security-test')
}

jacocoTestCoverageVerification {
    violationRules {
        rule {
            limit {
                minimum = 0.09
            }
        }
    }
}
