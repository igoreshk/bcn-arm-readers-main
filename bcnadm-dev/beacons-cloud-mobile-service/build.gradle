plugins {
    id 'beacons.common-service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-starter-oauth2')
    implementation('org.springframework.kafka:spring-kafka:2.5.1.RELEASE')
}

jacocoTestCoverageVerification {
    violationRules {
        rule {
            limit {
                minimum = 0.5
            }
        }
    }
}
