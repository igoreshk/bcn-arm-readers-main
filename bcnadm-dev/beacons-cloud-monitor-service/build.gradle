plugins {
    id 'beacons.common-service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-starter-oauth2')
    implementation('org.apache.commons:commons-math3:3.6.1')

    implementation('org.springframework.kafka:spring-kafka:2.5.1.RELEASE')
    implementation('org.apache.curator:curator-client:5.0.0') {
        exclude group: 'org.apache.zookeeper', module: 'zookeeper'
    }
    implementation('org.apache.curator:curator-x-async:5.0.0') {
        exclude group: 'org.apache.curator', module: 'curator-framework'
        exclude group: 'org.apache.zookeeper', module: 'zookeeper'
    }
    implementation group: 'org.apache.curator', name: 'curator-framework', version: "5.0.0"
    implementation group: 'org.apache.zookeeper', name: 'zookeeper', version: '3.6.1'
    implementation group: 'com.google.code.gson', name: 'gson', version: '2.8.6'

    implementation(group: 'io.dropwizard.metrics', name: 'metrics-core', version: '4.1.9')
    implementation(group: 'io.dropwizard.metrics', name: 'metrics-graphite', version: '4.1.9')

    implementation('org.geotools:gt-main:26.1')

    testImplementation('org.springframework.kafka:spring-kafka-test:2.5.1.RELEASE')
    testImplementation group: 'org.assertj', name: 'assertj-core', version: '3.16.1'
}

jacocoTestCoverageVerification {
    violationRules {
        rule {
            limit {
                minimum = 0.74
            }
        }
    }
}
