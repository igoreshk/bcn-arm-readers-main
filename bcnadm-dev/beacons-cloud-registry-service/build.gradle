plugins {
    id 'beacons.service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-starter-config')
    implementation('org.springframework.cloud:spring-cloud-starter-eureka-server:1.4.7.RELEASE') {
        exclude group: 'org.springframework.boot', module: 'spring-boot-starter-tomcat'
    }
}
