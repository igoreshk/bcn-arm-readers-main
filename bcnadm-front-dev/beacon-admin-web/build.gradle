plugins {
    id "com.github.node-gradle.node" version "2.2.4"
}

node {
    version = '12.13.0'
    download = true
    npmVersion = '6.12.0'
}

task installDependencies(type: NpmTask) {
    args = ['install']
}

task checkStyleSheets(type: NpmTask) {
    args = ['run', 'stylelint']
}
checkStyleSheets.dependsOn(installDependencies)

task checkMain(type: NpmTask) {
    args = ['run', 'lint-main']
}
checkMain.dependsOn(checkStyleSheets)

task checkTest(type: NpmTask) {
    args = ['run', 'lint-test']
}
checkTest.dependsOn(checkMain)

task test(type: NpmTask) {
    args = ['test']
}
test.dependsOn(checkTest)

// build whole frontend and
task build(type: NpmTask) {
    args = ['run', 'build']
}
build.dependsOn(test)

task start(type: NpmTask) {
    args = ['start']
}

task clean() {
    delete 'dist'
    delete 'build'
    delete 'out'
}
