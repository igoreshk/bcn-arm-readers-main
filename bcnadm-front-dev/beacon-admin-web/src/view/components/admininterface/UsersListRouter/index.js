import { USER_DIALOG, USERS_DELETE_DIALOG, USERS_LIST } from 'src/consts/RouteConsts';
import React from 'react';
import { Route } from 'react-router-dom';

import UsersList from '../UsersList';
import UserDialog from '../UserDialog';
import UserDeleteDialog from '../UserDeleteDialog';

const UsersListRouter = () => (
  <>
    <Route path={USERS_LIST} component={UsersList} />
    <Route exact path={USER_DIALOG} component={UserDialog} />
    <Route exact path={USERS_DELETE_DIALOG} component={UserDeleteDialog} />
  </>
);

export default UsersListRouter;
