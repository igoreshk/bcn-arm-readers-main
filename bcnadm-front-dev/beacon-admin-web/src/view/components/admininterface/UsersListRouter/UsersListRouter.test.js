import UsersListRouter from 'src/view/components/admininterface/UsersListRouter/index';
import React from 'react';
import { shallow } from 'enzyme/build/index';

const wrapper = shallow(<UsersListRouter />);

describe('UsersListRouter', () => {
  it('Check matching snapshot', () => expect(wrapper).toMatchSnapshot());
  it('Component has been rendered', () => expect(wrapper.length).toEqual(1));
});
