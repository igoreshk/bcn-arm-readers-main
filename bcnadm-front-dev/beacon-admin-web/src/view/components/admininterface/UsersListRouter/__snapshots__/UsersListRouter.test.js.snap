// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersListRouter Check matching snapshot 1`] = `
<Fragment>
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "type": [Function],
      }
    }
    path="/users"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/users/:id"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/users/:id/delete"
  />
</Fragment>
`;
